lock-api/src/main/java/com/palantir/lock/LockClient.java

```diff
@@ -66,6 +68,11 @@ public static LockClient of(String clientId) {
         Preconditions.checkArgument(!clientId.equals(INTERNAL_LOCK_GRANT_CLIENT_ID));
         return new LockClient(clientId);
     }
+    
+    @JsonCreator
+    public static LockClient valueOf(String clientId) {
```

Comment:
valueOf has other nice benefits, though unclear if will actually be of use... e.g. could use this as a jaxrs @PathParam or something... essentially jackson has some special things which look for a method named valueOf (or fromString)